python/sparkdl/transformers/tf_tensor.py

```diff
@@ -69,12 +113,19 @@ def _optimize_for_inference(self):
         input_node_names = [tfx.op_name(tnsr_name) for _, tnsr_name in input_mapping]
         output_node_names = [tfx.op_name(tnsr_name) for tnsr_name, _ in output_mapping]
 
+        # Build GraphFunction wrapping the model graph
+        graph_fn = GraphFunction(graph_def=gin.graph_def, input_names=input_node_names,
+                                 output_names=output_node_names)
+        # Inject cast ops to convert float64 input fed from Spark into the datatypes of the
+        # Graph's input nodes.
+        graph_fn_with_casts = self._addCastOps(graph_fn)
+
+        # Optimize for inference, replacing placeholders with float64
```

Comment:
do you remember what "optimize for inference" means here? (i forget)

Reply:
AFAIK it prunes out ops/nodes that aren't involved in computing the specified output tensors